src/components/MessagesList/MessagesGroup/Message/MessagePart/Reactions.vue

```diff
@@ -0,0 +1,246 @@
+		NcEmojiPicker,
+		NcPopover,
+		EmoticonOutline,
+	},
+
+	props: {
+		token: {
+			type: String,
+			required: true,
+		},
+		/**
+		 * Whether the current user can react to the message.
+		 */
+		canReact: {
+			type: Boolean,
+			default: false,
+		},
+		/**
+		 * The message id.
+		 */
+		id: {
+			type: [String, Number],
+			required: true,
+		},
+	},
+
+	emits: ['emoji-picker-toggled'],
+
+	setup() {
+		const guestNameStore = useGuestNameStore()
+		const reactionsStore = useReactionsStore()
+		return {
+			guestNameStore,
+			reactionsStore,
+		 }
+	},
+
+	data() {
+		return {
+			detailedReactionsLoading: false,
+		}
+	},
+
+	computed: {
+		hasReactions() {
+			return Object.keys(Object(this.detailedReactions)).length !== 0
+		},
+
+		detailedReactions() {
+			return this.reactionsStore.getReactions(this.token, this.id)
+		},
+
+		plainReactions() {
+			return this.$store.getters.message(this.token, this.id).reactions
+		},
+
+		reactionsSelf() {
+			return this.$store.getters.message(this.token, this.id).reactionsSelf
+		},
+
+		actorId() {
+			return this.$store.getters.getActorId()
+		},
+	},
+
+	methods: {
+		fetchReactions() {
+			if (!this.hasReactions) {
+				this.reactionsStore.fetchReactions(this.token, this.id)
+			}
+		},
+
+		userHasReacted(reaction) {
+			if (!this.reactionsSelf) {
+				return false
+			}
+			return this.reactionsSelf.find(emoji => emoji === reaction)
+		},
+
+		async handleReactionClick(clickedEmoji) {
+			if (!this.canReact) {
+				showError(t('spreed', 'No permission to post reactions in this conversation'))
+				return
+			}
+
+			// Check if current user has already added this reaction to the message
+			if (!this.userHasReacted(clickedEmoji)) {
+				this.$store.dispatch('addReactionToMessage', {
+					token: this.token,
+					messageId: this.id,
+					selectedEmoji: clickedEmoji,
+				})
+			} else {
+				this.$store.dispatch('removeReactionFromMessage', {
+					token: this.token,
+					messageId: this.id,
+					selectedEmoji: clickedEmoji,
+				})
+			}
+		},
+
+		getDisplayNameForReaction(reactingParticipant) {
+			if (reactingParticipant.actorType === ATTENDEE.ACTOR_TYPE.GUESTS) {
+				return this.guestNameStore.getGuestNameWithGuestSuffix(this.token, reactingParticipant.actorId)
+			}
+
+			const displayName = reactingParticipant.actorDisplayName.trim()
+			if (displayName === '') {
+				return t('spreed', 'Deleted user')
+			}
+
+			return displayName
+		},
+
+		reactionsCount(reaction) {
+			return this.detailedReactions[reaction]?.length ?? this.plainReactions[reaction]
+		},
+
+		getReactionSummary(reaction) {
+			// Check if the reaction details are loaded
+			if (!this.hasReactions) {
+				return ''
+			}
+			const list = this.detailedReactions[reaction]
+			const summary = []
+
+			for (const item in list) {
+				if (list[item].actorType === this.$store.getters.getActorType()
+					&& list[item].actorId === this.$store.getters.getActorId()) {
+					summary.unshift(t('spreed', 'You'))
```

Comment:
Not covered by test, thought very frequent to happen. Try https://test-utils.vuejs.org/api/#trigger